akka-remote/src/main/scala/akka/remote/transport/TestTransport.scala

```diff
@@ -243,7 +244,7 @@ object TestTransport {
       val originalBehavior = currentBehavior
 
       push(
-        (params: A) ⇒ for (delayed ← controlPromise.future; original ← originalBehavior(params)) yield original)
+        (params: A) ⇒ for (_ ← controlPromise.future; original ← originalBehavior(params)) yield original)
```

Comment:
This seems like a cumbersome way to write `controlPromise.flatMap(_ => originalBehavior(params))` - or is there a difference?

Reply:
Well, for this issue (the overarching one), since the changes touch multiple modules, many files, my goal is to retain as much original code, make as few changes and consider improvements like that out of scope.

Reply:
I'm not saying I like it this way. I did originally want to rewrite it. Let me see...

Reply:
I agree about being a bit conservative here, but this one caught my eye :). If it's nontrivial it might be better to leave it for another day / separate PR.